src/analyses/wrapperFunctionAnalysis.ml

```diff
@@ -0,0 +1,230 @@
+(** An analysis that handles the case when an interesting function is called
+    from a wrapper function all over the code. Currently handles the [malloc]-
+    family of memory allocation functions, as well as [pthread_create] *)
+
+open Prelude.Ana
+open Analyses
+open GobConfig
+open ThreadIdDomain
+module Q = Queries
+
+(* Functor argument for creating the chain lattice of unique calls *)
+module type UniqueCountArgs = sig
+  val unique_count : unit -> int
+  val label : string
+end
+
+(* Functor argument for determining wrapper and wrapped functions *)
+module type WrapperArgs = sig
+  val wrappers : unit -> string list
+  val is_wrapped : LibraryDesc.special -> bool
+end
+
+(* The main analysis, generic to which functions are being wrapped. *)
+module SpecBase (UniqueCountArgs : UniqueCountArgs) (WrapperArgs : WrapperArgs) =
+struct
+  include Analyses.DefaultSpec
```

Comment:
Might as well switch to `IdentitySpec` here to avoid all the identity transfer functions.